index.js

```diff
@@ -0,0 +1,138 @@
+const express = require('express');
+const server = express();
+const cors = require('cors');
+
+const db = require('./database/dbConfig.js');
+const bcrypt = require('bcryptjs');
+const session = require('express-session');
+//configure express-session middleware
+const KnexSessionStore = require('connect-session-knex')(session);
+  const sessionConfig = {
+    name: 'notsession', // default is connect.sid
+    secret: 'nobody tosses a dwarf!',
+    cookie: {
+      maxAge: 1 * 24 * 60 * 60 * 1000,
+      secure: false, // only set cookies over https. Server will not send back a cookie over http.
+    }, // 1 day in milliseconds
+    httpOnly: true, // don't let JS code access cookies. Browser extensions run JS code on your browser!
+    resave: false,
+    saveUninitialized: false,
+    store: new KnexSessionStore({
+      tablename: 'sessions',
+      sidfieldname: 'sid',
+      knex: db,
+      createtable: true,
+      clearinterval: 100 * 60 *60,
+    })
+  };
+server.use(session(sessionConfig))
+
+
+
+
+server.use(express.json());
+
+server.use(cors());
+
+function protected(req, res, next) {
+  if (req.session && req.session.username) {
+    next();
+  } else {
+    res.status(401).json({ message: 'you shall not pass!!' });
+  }
+}
+
+
+server.get('/', (req, res) => {
+  res.send('Its Alive!');
+});
+
+server.post('/api/register', (req,res) =>{
+  //grab credentials
+  const creds = req.body
+  //hash the password
+  const hash = bcrypt.hashSync(creds.password, 3); //really 2^3, hashed 8 times
+
+  //replace user password with the hash
+  creds.password = hash;
+  //save the user
+
+  db('users').insert(creds).then(ids => {
+
+    const id = ids[0];
+  res.status(201).json(id);
+
+  }).catch(err => res.status(500).send(err));
```

Comment:
I'd recommend console.logging the error in your `catch()` block. The error tends to be shortened in PostMan, and conceptually, it's not the client's job to fix your server, so they don't really need to know the exact error.

```js
// snippet
.catch(err => {
    console.log('/api/register POST error:', err);
    res.status(500).send('Please try again later.');
});
```